test/metrics/metrics_test.go

```diff
@@ -0,0 +1,311 @@
+	VerifyMemberMetricsService(t, memberAwait)
+	metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName})
+	usersignups := map[string]*toolchainv1alpha1.UserSignup{}
+
+	// when
+	for i := 1; i <= 3; i++ {
+		username := fmt.Sprintf("user-%04d", i)
+		usersignups[username] = CreateAndApproveSignup(t, hostAwait, username, memberAwait.ClusterName)
+		for j := 1; j < i; j++ { // deactivate and reactivate as many times as necessary (based on its "number")
+			// deactivate the user
+			_, err := hostAwait.UpdateUserSignupSpec(usersignups[username].Name, func(usersignup *toolchainv1alpha1.UserSignup) {
+				states.SetDeactivated(usersignup, true)
+			})
+			require.NoError(t, err)
+			err = hostAwait.WaitUntilMasterUserRecordDeleted(username)
+			require.NoError(t, err)
+			// reactivate the user
+			CreateAndApproveSignup(t, hostAwait, username, memberAwait.ClusterName, WithIdentityID(usersignups[username].Spec.Userid))
+		}
+	}
+
+	// then verify the value of the `sandbox_users_per_activations` metric
+	metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 1, "activations", "1", "domain", "external") // 1 activation
+	metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 1, "activations", "2", "domain", "external") // 1 activation
+	metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 1, "activations", "3", "domain", "external") // 1 activation
+
+	t.Run("restart host-operator pod and verify that metrics are still available", func(t *testing.T) {
+		// given
+		metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName})
+
+		// when deleting the host-operator pod to emulate an operator restart during redeployment.
+		err := hostAwait.DeletePods(client.InNamespace(hostAwait.Namespace), client.MatchingLabels{"name": "controller-manager"})
+
+		// then check how much time it takes to restart and process all existing resources
+		require.NoError(t, err)
+		// host metrics should become available again at this point
+		_, err = hostAwait.WaitForRouteToBeAvailable(hostAwait.Namespace, "host-operator-metrics-service", "/metrics")
+		require.NoError(t, err, "failed while setting up or waiting for the route to the 'host-operator-metrics-service' service to be available")
+		// also verify that the metric values "survived" the restart
+		metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "1", "domain", "external") // user-0001 was 1 time (unchanged after pod restarted)
+		metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "2", "domain", "external") // user-0002 was 2 times (unchanged after pod restarted)
+		metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "3", "domain", "external") // user-0003 was 3 times (unchanged after pod restarted)
+	})
+}
+
+func TestMetricsWhenUsersDeleted(t *testing.T) {
```

Comment:
could we rename it for example to:
```suggestion
func TestUsersPerActivationsMetricsStaysTheSameWhenUsersDeleted(t *testing.T) {
```

Reply:
I'm not a fan of such long names TBH

Reply:
it's self-explanatory.... 
or you could add a comment to the test.
either a comment or the self-explanatory test name would really help :pray:

Reply:
done in aab86da